test/onnx/parse/dynamicquantizelinear_simplfy_2d_test.cpp

```diff
@@ -0,0 +1,84 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <onnx_test.hpp>
+#include <migraphx/common.hpp>
+#include <migraphx/simplify_dynamicquantizelinear.hpp>
+
+TEST_CASE(dynamicquantizelinear_simplify_2d_test)
+{
+    migraphx::program p;
+    auto* mm    = p.get_main_module();
+    auto x_dims = {3, 3};
+    auto x_type = migraphx::shape::float_type;
+    auto x      = mm->add_parameter("x", {x_type, x_dims});
+    auto l0     = mm->add_literal({0.f});
+
+    std::vector<size_t> axes(x->get_shape().lens().size());
+    std::iota(axes.begin(), axes.end(), 0);
+
+    auto reduce_max_x = mm->add_instruction(migraphx::make_op("reduce_max", {{"axes", axes}}), x);
+    auto max_x        = add_common_op(*mm, migraphx::make_op("max"), {l0, reduce_max_x});
+
+    auto reduce_min_x = mm->add_instruction(migraphx::make_op("reduce_min", {{"axes", axes}}), x);
+    auto min_x        = add_common_op(*mm, migraphx::make_op("min"), {l0, reduce_min_x});
+
+    auto q_range = mm->add_literal(migraphx::literal{
+        migraphx::shape{x_type, max_x->get_shape().lens()},
+        {std::numeric_limits<int8_t>::max() - std::numeric_limits<int8_t>::min()}});
+    auto q_min   = mm->add_literal(migraphx::literal{
+        migraphx::shape{x_type, max_x->get_shape().lens()}, {std::numeric_limits<int8_t>::min()}});
+    auto q_max   = mm->add_literal(migraphx::literal{
+        migraphx::shape{x_type, min_x->get_shape().lens()}, {std::numeric_limits<int8_t>::max()}});
+
+    auto sub0    = mm->add_instruction(migraphx::make_op("sub"), max_x, min_x);
+    auto y_scale = mm->add_instruction(migraphx::make_op("div"), sub0, q_range);
+
+    auto div1      = add_common_op(*mm, migraphx::make_op("div"), {min_x, y_scale});
+    auto interm_zp = add_common_op(*mm, migraphx::make_op("sub"), {q_min, div1});
+
+    auto saturate     = mm->add_instruction(migraphx::make_op("clip"), interm_zp, q_min, q_max);
+    auto round        = mm->add_instruction(migraphx::make_op("nearbyint"), saturate);
+    auto y_zero_point = mm->add_instruction(
+        migraphx::make_op("convert", {{"target_type", migraphx::shape::int8_type}}), round);
+
+    auto scale_y_bcast =
+        mm->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_dims}}), y_scale);
+
+    auto y_pt_c_bcast = mm->add_instruction(
+        migraphx::make_op("multibroadcast", {{"out_lens", x_dims}}), y_zero_point);
+
+    auto y_out =
+        mm->add_instruction(migraphx::make_op("quantizelinear"), x, scale_y_bcast, y_pt_c_bcast);
+
+    auto sub = mm->add_instruction(migraphx::make_op("sub"), y_out, y_out);
+    mm->add_instruction(migraphx::make_op("dot"), y_out, sub);
+
+    auto prog = optimize_onnx("dynamicquantizelinear_2d_dot_test.onnx");
+    auto* m   = prog.get_main_module();
+    migraphx::run_passes(*m, {migraphx::simplify_dynamicquantizelinear{}});
```

Comment:
You are testing results for the pass which should go under `test/simplify_dynamicquantizelinear_test.cpp`.
under `/test/onnx/parse` it should test whether parsing is correct ornot.

Reply:
This was done to verify a pass after parse yields a predictable result as we don't support uint8 x int8 quantization.